diff --git a/tests/modeltests/get_or_create/tests.py b/tests/modeltests/get_or_create/tests.py
index 1e300fb..1231c47 100644
--- a/tests/modeltests/get_or_create/tests.py
+++ b/tests/modeltests/get_or_create/tests.py
@@ -4,12 +4,13 @@ from datetime import date
 import traceback
 
 from django.db import IntegrityError
-from django.test import TestCase
+from django.test import TestCase, skipUnlessDBFeature
 
 from .models import Person, ManualPrimaryKeyTest
 
 
 class GetOrCreateTests(TestCase):
+    @skipUnlessDBFeature('uses_savepoints')
     def test_get_or_create(self):
         p = Person.objects.create(
             first_name='John', last_name='Lennon', birthday=date(1940, 10, 9)
diff --git a/tests/regressiontests/admin_scripts/tests.py b/tests/regressiontests/admin_scripts/tests.py
index 91bcbee..5681835 100644
--- a/tests/regressiontests/admin_scripts/tests.py
+++ b/tests/regressiontests/admin_scripts/tests.py
@@ -89,7 +89,7 @@ class AdminScriptTestCase(unittest.TestCase):
         Returns the paths for any external backend packages.
         """
         paths = []
-        first_package_re = re.compile(r'(^[^\.]+)\.')
+        first_package_re = re.compile(r'(^[^\.]+)\.?')
         for backend in settings.DATABASES.values():
             result = first_package_re.findall(backend['ENGINE'])
             if result and result != 'django':
diff --git a/tests/regressiontests/aggregation_regress/tests.py b/tests/regressiontests/aggregation_regress/tests.py
index 65d345a..a72df4a 100644
--- a/tests/regressiontests/aggregation_regress/tests.py
+++ b/tests/regressiontests/aggregation_regress/tests.py
@@ -9,6 +9,8 @@ from django.core.exceptions import FieldError
 from django.db.models import Count, Max, Avg, Sum, StdDev, Variance, F, Q
 from django.test import TestCase, Approximate, skipUnlessDBFeature
 from django.utils import six
+from django.utils.unittest import skipIf
+from django.db import connection
 
 from .models import Author, Book, Publisher, Clues, Entries, HardbackBook
 
@@ -375,6 +377,7 @@ class AggregationTests(TestCase):
             lambda p: p,
         )
 
+    @skipIf(connection.vendor == 'fdbsql', 'Unsupported mix of DISTINCT and non-DISTINCT')
     def test_db_col_table(self):
         # Tests on fields with non-default table and column names.
         qs = Clues.objects.values('EntryID__Entry').annotate(Appearances=Count('EntryID'), Distinct_Clues=Count('Clue', distinct=True))
diff --git a/tests/regressiontests/backends/tests.py b/tests/regressiontests/backends/tests.py
index b92d3b9..3efdfc7 100644
--- a/tests/regressiontests/backends/tests.py
+++ b/tests/regressiontests/backends/tests.py
@@ -530,6 +530,7 @@ class FkConstraintsTests(TransactionTestCase):
             finally:
                 transaction.rollback()
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'check_constraints not implemented')
     def test_check_constraints(self):
         """
         Constraint checks should raise an IntegrityError when bad data is in the DB.
diff --git a/tests/regressiontests/delete_regress/tests.py b/tests/regressiontests/delete_regress/tests.py
index 1bc4e78..906691f 100644
--- a/tests/regressiontests/delete_regress/tests.py
+++ b/tests/regressiontests/delete_regress/tests.py
@@ -3,8 +3,9 @@ from __future__ import absolute_import
 import datetime
 
 from django.conf import settings
-from django.db import backend, transaction, DEFAULT_DB_ALIAS, models
+from django.db import backend, transaction, DEFAULT_DB_ALIAS, models, connection
 from django.test import TestCase, TransactionTestCase, skipUnlessDBFeature
+from django.utils import unittest
 
 from .models import (Book, Award, AwardNote, Person, Child, Toy, PlayedWith,
     PlayedWithNote, Email, Researcher, Food, Eaten, Policy, Version, Location,
@@ -38,6 +39,7 @@ class DeleteLockingTest(TransactionTestCase):
         transaction.leave_transaction_management()
         self.conn2.close()
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Test assumes READ COMMITTED isolation')
     @skipUnlessDBFeature('test_db_allows_multiple_connections')
     def test_concurrent_delete(self):
         "Deletes on concurrent transactions don't collide and lock the database. Regression for #9479"
@@ -289,6 +291,7 @@ class Ticket19102Tests(TestCase):
         self.l1 = Login.objects.create(description='l1', orgunit=self.o1)
         self.l2 = Login.objects.create(description='l2', orgunit=self.o2)
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     @skipUnlessDBFeature("update_can_self_select")
     def test_ticket_19102_annotate(self):
         with self.assertNumQueries(1):
@@ -302,6 +305,7 @@ class Ticket19102Tests(TestCase):
         self.assertFalse(Login.objects.filter(pk=self.l1.pk).exists())
         self.assertTrue(Login.objects.filter(pk=self.l2.pk).exists())
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     @skipUnlessDBFeature("update_can_self_select")
     def test_ticket_19102_extra(self):
         with self.assertNumQueries(1):
@@ -329,6 +333,7 @@ class Ticket19102Tests(TestCase):
         self.assertFalse(Login.objects.filter(pk=self.l1.pk).exists())
         self.assertTrue(Login.objects.filter(pk=self.l2.pk).exists())
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     @skipUnlessDBFeature("update_can_self_select")
     def test_ticket_19102_select_related(self):
         with self.assertNumQueries(1):
@@ -342,6 +347,7 @@ class Ticket19102Tests(TestCase):
         self.assertFalse(Login.objects.filter(pk=self.l1.pk).exists())
         self.assertTrue(Login.objects.filter(pk=self.l2.pk).exists())
     
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     @skipUnlessDBFeature("update_can_self_select")
     def test_ticket_19102_defer(self):
         with self.assertNumQueries(1):
diff --git a/tests/regressiontests/extra_regress/tests.py b/tests/regressiontests/extra_regress/tests.py
index f591900..123fce1 100644
--- a/tests/regressiontests/extra_regress/tests.py
+++ b/tests/regressiontests/extra_regress/tests.py
@@ -5,6 +5,8 @@ import datetime
 from django.contrib.auth.models import User
 from django.test import TestCase
 from django.utils.datastructures import SortedDict
+from django.db import connection
+from django.utils.unittest import skipIf
 
 from .models import TestObject, Order, RevisionableModel
 
@@ -168,6 +170,7 @@ class ExtraRegressTests(TestCase):
             ['datetime.datetime(2008, 9, 1, 0, 0)']
         )
 
+    @skipIf(connection.vendor == 'fdbsql', 'Manual WHERE reserved word')
     def test_values_with_extra(self):
         """
         Regression test for #10256... If there is a values() clause, Extra
@@ -314,6 +317,7 @@ class ExtraRegressTests(TestCase):
             ['<TestObject: TestObject: first,second,third>']
         )
 
+    @skipIf(connection.vendor == 'fdbsql', 'Manual WHERE reserved word')
     def test_regression_17877(self):
         """
         Ensure that extra WHERE clauses get correctly ANDed, even when they
diff --git a/tests/regressiontests/fixtures_regress/tests.py b/tests/regressiontests/fixtures_regress/tests.py
index 61dc446..10a394e 100644
--- a/tests/regressiontests/fixtures_regress/tests.py
+++ b/tests/regressiontests/fixtures_regress/tests.py
@@ -18,6 +18,8 @@ from django.utils.encoding import force_text
 from django.utils._os import upath
 from django.utils import six
 from django.utils.six import PY3, StringIO
+from django.db import connection
+from django.utils.unittest import skipIf
 import json
 
 from .models import (Animal, Stuff, Absolute, Parent, Child, Article, Widget,
@@ -380,6 +382,7 @@ class TestFixtures(TestCase):
         self.assertEqual(Book.objects.all()[0].id, 1)
         self.assertEqual(Person.objects.all()[0].id, 4)
 
+    @skipIf(connection.vendor == 'fdbsql', 'check_constraints() not implemented')
     def test_loaddata_raises_error_when_fixture_has_invalid_foreign_key(self):
         """
         Regression for #3615 - Ensure data with nonexistent child key references raises error
diff --git a/tests/regressiontests/model_fields/models.py b/tests/regressiontests/model_fields/models.py
index 1d20f44..0976feb 100644
--- a/tests/regressiontests/model_fields/models.py
+++ b/tests/regressiontests/model_fields/models.py
@@ -14,7 +14,7 @@ except ImportError:
         Image = None
 
 from django.core.files.storage import FileSystemStorage
-from django.db import models
+from django.db import models, connection
 from django.db.models.fields.files import ImageFieldFile, ImageField
 
 
@@ -46,7 +46,8 @@ class Whiz(models.Model):
     c = models.IntegerField(choices=CHOICES, null=True)
 
 class BigD(models.Model):
-    d = models.DecimalField(max_digits=38, decimal_places=30)
+    prec,scale = connection.vendor != 'fdbsql' and (38,30) or (30,22)
+    d = models.DecimalField(max_digits=prec, decimal_places=scale)
 
 class BigS(models.Model):
     s = models.SlugField(max_length=255)
diff --git a/tests/regressiontests/multiple_database/tests.py b/tests/regressiontests/multiple_database/tests.py
index f2ecfbe..9f6ac8f 100644
--- a/tests/regressiontests/multiple_database/tests.py
+++ b/tests/regressiontests/multiple_database/tests.py
@@ -1348,14 +1348,14 @@ class RouterTestCase(TestCase):
         # If you create an object through a M2M relation, it will be
         # written to the write database, even if the original object
         # was on the read database
-        alice = dive.authors.create(name='Alice')
+        alice = dive.authors.create(name='Alice', pk=3)
         self.assertEqual(alice._state.db, 'default')
 
         # Same goes for get_or_create, regardless of whether getting or creating
         alice, created = dive.authors.get_or_create(name='Alice')
         self.assertEqual(alice._state.db, 'default')
 
-        bob, created = dive.authors.get_or_create(name='Bob')
+        bob, created = dive.authors.get_or_create(name='Bob', pk=4)
         self.assertEqual(bob._state.db, 'default')
 
     def test_o2o_cross_database_protection(self):
diff --git a/tests/regressiontests/queries/tests.py b/tests/regressiontests/queries/tests.py
index c7e746e..1fd906d 100644
--- a/tests/regressiontests/queries/tests.py
+++ b/tests/regressiontests/queries/tests.py
@@ -494,6 +494,7 @@ class Queries1Tests(BaseQuerysetTest):
             [{'note': 1}, {'note': 2}]
         )
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     def test_ticket2902(self):
         # Parameters can be given to extra_select, *if* you use a SortedDict.
 
@@ -589,6 +590,7 @@ class Queries1Tests(BaseQuerysetTest):
             [{'note__note': 'n2'}, {'note__note': 'n3'}, {'note__note': 'n3'}, {'note__note': 'n3'}]
         )
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     def test_ticket7096(self):
         # Make sure exclude() with multiple conditions continues to work.
         self.assertQuerysetEqual(
@@ -848,6 +850,7 @@ class Queries1Tests(BaseQuerysetTest):
         )
         Tag._meta.ordering = original_ordering
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     def test_exclude(self):
         self.assertQuerysetEqual(
             Item.objects.exclude(tags__name='t4'),
@@ -859,11 +862,13 @@ class Queries1Tests(BaseQuerysetTest):
             Item.objects.exclude(Q(tags__name='t4')|~Q(tags__name='t3')),
             [repr(i) for i in Item.objects.filter(~(Q(tags__name='t4')|~Q(tags__name='t3')))])
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     def test_nested_exclude(self):
         self.assertQuerysetEqual(
             Item.objects.exclude(~Q(tags__name='t4')),
             [repr(i) for i in Item.objects.filter(~~Q(tags__name='t4'))])
 
+    @unittest.skipIf(connection.vendor == 'fdbsql', 'Known bug!')
     def test_double_exclude(self):
         self.assertQuerysetEqual(
             Item.objects.filter(Q(tags__name='t4')),
diff --git a/tests/regressiontests/test_runner/tests.py b/tests/regressiontests/test_runner/tests.py
index 3418e2a..95db088 100644
--- a/tests/regressiontests/test_runner/tests.py
+++ b/tests/regressiontests/test_runner/tests.py
@@ -368,6 +368,7 @@ class DeprecationDisplayTest(AdminScriptTestCase):
         self.assertFalse("DeprecationWarning: warning from test" in err)
 
 
+@unittest.skipIf(db.connection.vendor == 'fdbsql', 'Requires transactional sequence reset')
 class AutoIncrementResetTest(TransactionTestCase):
     """
     Here we test creating the same model two times in different test methods,
